statechart/models/statechart_mixin.py

```diff
@@ -236,21 +224,42 @@ def fields_view_get(self, view_id=None, view_type='form',
 @api.model
 def _sc_patch(self):
     cls = type(self)
+
     if getattr(cls, '_sc_patch_done', False):
         return
+    cls._sc_patch_done = True
```

Comment:
@sbidoul If a parent without statechart is marked as patched, a child with a statechart will nerver be patched.

Reply:
Well, that's not what I observed, because the particular `cls` we have here is not inherited. The class that is actually inherited is `cls.__bases__[0]`. But you should check it with a debugger, the class model is not obvious at first glance (nor on the second glance, and it's entirely possible I've overlooked something :)